### ISSUE TITLE
Feature request: adding more versatility in the tfp.math.psd_kernels 

### ISSUE THREAD

**Issue by FelixRb96** [tensorflow/probability#1131]
Hello, 

first of all I really appreciate the effort everybody is putting in. However for my latest application of gaussian processes it would be very beneficial to have a wider choice of kernels in the module mentioned in the title, maybe some other people have encountered the same issue. For some applications with gaussian processes it is very beneficial to actually estimate the Whittle-Matern nu parameter since it directly controls the smoothness of the process. I see the point that implementation is far more tedious for the general case but having a bit of an R background it is a nice feature to estimate the smoothness parameter too. 

Thank you already for your time !

**Comment by brianwa84** [comment]
How timely!! Just yesterday @srvasude committed the necessary bessel functions. :)

We were just brainstorming generalizing the Matern family a week or two ago.
As they currently stand, the bessel's don't have derivatives wrt nu, so you couldn't use backprop, but could do a grid search or s.t.

**Comment by srvasude** [comment]
Hi,
  I am going to close this issue as I've added a `GeneralizedMatern` kernel which should tackle this case. As Brian has mentioned, the bessel functions don't have derivatives with respect to `nu`, so optimization with respect to that parameter has to be gradient free. If this is something you need, feel free to reopen this issue (and we'll change the title since this affects other distributions that rely on Bessel functions).
